hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHAWithData.java

```diff
@@ -104,6 +106,43 @@ public void testMultipartUpload() throws Exception {
     testMultipartUploadWithOneOmNodeDown();
   }
 
+  @Test
+  public void testOMHAMetrics()
+      throws InterruptedException, TimeoutException {
+    waitForLeaderToBeReady();
+
+    OzoneManager leaderOM = getCluster().getOMLeader();
+    OzoneManager randomOM = getCluster().getOzoneManager(1);
+
+    // Get OMHAMetrics
+    OMHAMetrics omhaMetrics = randomOM.getOmhaMetrics();
+
+    if (randomOM.getOMNodeId()
+        .equals(leaderOM.getOMNodeId())) {
+      Assertions.assertEquals(1L,
+          omhaMetrics.getOmhaInfoOzoneManagerHALeaderState());
+    } else {
+      Assertions.assertEquals(0L,
+          omhaMetrics.getOmhaInfoOzoneManagerHALeaderState());
+    }
```

Comment:
Instead of selecting a "random" node and having only a single assertion, the test could iterate all OMs and verify value in each.

Also, please reduce duplication:

```suggestion
    int expected;
    if (randomOM.getOMNodeId()
        .equals(leaderOM.getOMNodeId())) {
      expected = 1;
    } else {
      expected = 0;
    }
    
    Assertions.assertEquals(expected,
        omhaMetrics.getOmhaInfoOzoneManagerHALeaderState());
```

(Please feel free to replace the `if-else` with `?:`.)